### PR TITLE
Bug 1139517 - Changes log name of buildbot logs from 'builds-4h' to 'buildbot_text'

### PR DESCRIPTION
**Pull request by mister-raindrop**
This commit changes all the references to 'builds-4h' to 'buildbot_text'. Following
are the changed files along with the no. of occurences that have been changed in
each.

1. tests/etl/test_buildapi.py: 3 occurences
2. tests/sample_data/job_data.txt: 304 occurences
3. treeherder/etl/buildapi.py: 1 occurence
4. treeherder/model/sample_data/job_data.json.sample: 2 occurences

In treeherder/webapp/api/logslice.py, a conditional was removed. The todo above
it instructed it to be removed once this bug was addressed.

To make sure all tests run properly, three files were renamed. Only the portion
of the filename that said 'builds-4h' was changed to say 'buildbot_text'.

Test results after applying these changes: http://pastebin.com/4rv0mtAK

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/471)

### CODE REVIEW

File: treeherder/webapp/api/logslice.py [hunk 1]
```diff
@@ -59,10 +59,6 @@ def list(self, request, project, jm):
         if start_line >= end_line:
             return Response("``end_line`` must be larger than ``start_line``", 400)
 
-        # @todo: remove once Bug 1139517 is addressed
```
Comment (camd) [inline treeherder/webapp/api/logslice.py]: This comment (I think I put it in there) doesn't quite tell the whole story.  When new logs are ingested, they will go in with the new names.  But we will still have at least 3 months of data with the logs that have the old name.  So, in the interim, we need to handle both names.

we should still remove this set of lines.  But then add checking for both later...

File: treeherder/webapp/api/logslice.py [hunk 2]
```diff
@@ -59,10 +59,6 @@ def list(self, request, project, jm):
...
-        if log_name == "buildbot_text":
-            log_name = 'builds-4h'
-
         # get only the log that matches the ``log_name``
         logs = jm.get_log_references(job_id)
```
Comment (camd) [inline treeherder/webapp/api/logslice.py]: in the following ``log = next(log for log in logs if log["name"] == log_name)`` it should be changed to something like:

    # @todo: remove after no more logs named 'builds-4h' exist in the db.  Should be after Aug 30, 2015.
    log = next(log for log in logs if log["name"] in [log_name, 'builds-4h'])

Comment (camd) [inline treeherder/webapp/api/logslice.py]: Ed tells me it's 4 months until old data is expunged.  So that date above should be right.

**Comment by mister-raindrop** [comment]
Added the check as you suggested. I didn't take into account the old data in the db before. I hope there aren't any other changes than this, especially for old logs?

**Comment by camd** [comment]
This looks good.  I'm going to test it locally to see how it does.  Thanks!

**Comment by camd** [comment]
@mister-raindrop : would you be so kind as to rebase this on the latest master?  We just changed the name of the repo from ``treeherder-service`` to just ``treeherder``.  There are lots of strings that were changed in that process, so I'm having trouble running your branch locally.

Sorry for the hassle, but you should only have to make this change once.  We're not changing again.  :)

**Comment by mister-raindrop** [comment]
@camd : Hey, I pulled the changes on my master and merged them from upstream. Switched to my branch, ran 'git rebase master' and pushed it again with -f. Was that right? Do I need to change my remotes or anything else?

**Comment by camd** [comment]
This is looking very good.  I want to do a bit more testing with it before I merge though.  Testing backward compatibility.  I am on PTO tomorrow (Friday) but will test if I'm able.  I may not get to merge till Monday, however.  Sorry for the delay.

**Comment by mister-raindrop** [comment]
Alright, no problem. I guess I didn't screw up with the rebase then :3

I'll check back again on Monday. Thanks.

**Pull request merged**